### DISCUSSION TITLE
On Field - add my Culture Info ?

### DISCUSSION THREAD

**Discussion by mrcorale** [Megabit/Blazorise#3408] [Q&A] [answered]
Hello, it is possible to add System.Globalization.CultureInfo.GetCultureInfo on Field there im typing the value, Thank's ?
![Screenshot_1](https://user-images.githubusercontent.com/52911474/152687748-ec730c68-50e1-463e-b7ca-5ba5c41bb640.jpg)

**Comment by stsrki** [comment, marked as answer]
Can you please explain why you need that CultureInfo on an input?

**Reply by mrcorale** [reply]
hi, 
Hi, I need CultureInfo for Users who have a different browser language than RO, I currently use Blazorise v. 0.9.2.5 I can't put another higher version, because it's another version of net framework

**Reply by mrcorale** [reply]
I implemented this @code in Starup at Blazor, but nothing helps:

```
services.Configure<RequestLocalizationOptions>(options =>
            {
                var supportedCultures = new[] { new CultureInfo("ro-RO") };
                options.DefaultRequestCulture = new RequestCulture("ro-RO");
                options.SupportedCultures = supportedCultures;
                options.SupportedUICultures = supportedCultures;

                CultureInfo customCulture = (CultureInfo)Thread.CurrentThread.CurrentCulture.Clone();
                Thread.CurrentThread.CurrentUICulture = CultureInfo.GetCultureInfo("ro-RO");
                var currentCulture = Thread.CurrentThread.CurrentCulture.Name;
                var ci = new CultureInfo(currentCulture)
                {
                    NumberFormat = { NumberDecimalSeparator = ",", NumberGroupSeparator = "." },
                    DateTimeFormat = { DateSeparator = "." }
                };
                Thread.CurrentThread.CurrentCulture = ci;
                Thread.CurrentThread.CurrentUICulture = ci;
            });
```

and in

```
public void Configure(IApplicationBuilder app, IWebHostEnvironment env,
                              ILogger<Startup> logger)
``` 

```
var options = app.ApplicationServices.GetService<IOptions<RequestLocalizationOptions>>();
            var cookieProvider = options.Value.RequestCultureProviders
                .OfType<CookieRequestCultureProvider>()
                .First();
            var urlProvider = options.Value.RequestCultureProviders
                .OfType<QueryStringRequestCultureProvider>().First();
            cookieProvider.Options.DefaultRequestCulture = new RequestCulture("ro-RO");
            urlProvider.Options.DefaultRequestCulture = new RequestCulture("ro-RO");
            cookieProvider.CookieName = "UserCulture";
            options.Value.RequestCultureProviders.Clear();
            options.Value.RequestCultureProviders.Add(cookieProvider);
            options.Value.RequestCultureProviders.Add(urlProvider);
            app.UseRequestLocalization(options.Value);
```

and on the Blazor page where the fields I need the information are. This one:

```
var cultureInfo = CultureInfo.GetCultureInfo("ro-RO");
            Thread.CurrentThread.CurrentCulture = cultureInfo;
            Thread.CurrentThread.CurrentUICulture = cultureInfo;
            CultureInfo.DefaultThreadCurrentCulture = cultureInfo;
            CultureInfo.DefaultThreadCurrentUICulture = cultureInfo;
            Thread.CurrentThread.CurrentUICulture = CultureInfo.GetCultureInfo("ro-RO");
            var currentCulture = Thread.CurrentThread.CurrentCulture.Name;
            var ci = new CultureInfo(currentCulture)
            {
                NumberFormat = { NumberDecimalSeparator = ",", NumberGroupSeparator = "." },
                DateTimeFormat = { DateSeparator = "." }
            };
            Thread.CurrentThread.CurrentCulture = ci;
            Thread.CurrentThread.CurrentUICulture = ci;
```


and it still doesn't give me the desired language at the interface, in the code it results in RO language but in Field NO, can you please help me?

**Reply by stsrki** [reply]
Closest to what you want is the `CurrentCultureInfo` that we have on the `NumericEdit` component. No other component has it. And it is used for parsing the value.

But even if do add `CurrentCultureInfo` to DateEdit and other components we can do it only in v1.x. No 0.9.x. Especially not 0.9.2 since we even don't have that branch to work with. The oldest version that we have is 0.9.4.

**Reply by mrcorale** [reply]
I know that in the new version from Blazorise is the possibility given with CultureInfo, but I have problems with net Framework 5 or another newer version, because in the company there are many users who use Windows XP.
That's why I use Net Core 3.1
That's why I addressed you with this question, because I can't finish solving the question that came to me !!!!

**Comment by stsrki** [comment]
😮 They're still using Win XP?

I don't think there is a solution here. And if there is, we can only do it in later versions. 

You can try setting the culture in the HTML if you already didn't https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Language. I cannot think of another way.

**Reply by mrcorale** [reply]
But is it possible for Blazorise to have a script or something like that telling me to change the language in Browser?

**Reply by stsrki** [reply]
You could add that I think with some custom components and/or JavaScript. There is nothing in Blazorise that could do that.

Or are you wanting to change the user language in the UI? https://blazorise.com/docs/helpers/localization

**Reply by mrcorale** [reply]
I understand, thank you very much, and sorry for the inconvenience!
